### ISSUE TITLE
Add a matchmaking form to the `/play` page

### ISSUE THREAD

**Issue by seanmturley** [seanmturley/common-ground#41]
# Description

Add a basic matchmaking form to the `/play` page. This should include:
- [x] Sensible default values for `format` and `match_type`
- [x] A "find match" `submit` button that makes a `POST` to the `/api/matchmaking` route handler

Consider that [useActionState](https://react.dev/reference/react/useActionState) may be appropriate.

**Comment by seanmturley** [comment]
The basics of this are in place, using `useActionState`, though the following remains to be done before this is at all functional:

- Adding the database tables and edge functions for matchmaking
- Subscribing to database changes to listen for when a match is found (#42)
- Consider hiding `format` and `match_type` options in the matchmaking form, since the plan would be to test/launch with only e.g. Bo1 Historic Pauper
- Styling

**Comment by seanmturley** [comment]
The `fetch` call requires an absolute URL, so NEXT_PUBLIC_BASE_URL was added to `.env.local` and included at the base of the URL for the call to the matchmaking route handler i.e.:
```
const response = await fetch(
    `${process.env.NEXT_PUBLIC_BASE_URL}/api/matchmaking`,

      ...

  );
```

**Comment by seanmturley** [comment]
Consider reworking the `JoinQueueForm` such that the database manipulations occur directly in the `formAction` function. This pattern was already implemented for the "cancel" button in #59, and could be applied here to simplify and centralize the logic.

**Comment by seanmturley** [comment]
Reworked as described in previous comment - the database manipulations for adding a player to the `queue` now occur directly in the `formAction` rather than an additional utility function. This simplifies and centralizes the logic for this functionality.

The `JoinQueueForm` was also renamed to simply `JoinQueue`, with some additional renaming of variables relating to the `useActionState`.
